views/components/ebulletin.njk

```diff
@@ -0,0 +1,99 @@
+{% from "components/forms-new.njk" import formErrors, formField, inputText %}
+
+{% macro renderField(field) %}
+    {% set fieldErrors = formHelpers.errorsForField(field, errors) %}
+    {{ formField(formHelpers.withFieldValue(field, data), fieldErrors) }}
+{% endmacro %}
+
+{% macro ebulletinForm() %}
+<form class="form-ebulletin" method="post" action="/about/ebulletin">
+    <h2 class="t1 t--underline accent--pink a--text">{{ __('toplevel.ebulletin.title') }}</h2>
+    <p class="form-ebulletin__intro">{{ __('toplevel.ebulletin.intro') }}</p>
+
+    {% if status === formHelpers.FORM_STATES.SUBMISSION_SUCCCESS %}
+        <h3 class="t5">{{ __('toplevel.ebulletin.responses.success.title') }}</h3>
+        <p>{{ __('toplevel.ebulletin.responses.success.body') | safe }}</p>
+    {% elseif status ===  formHelpers.FORM_STATES.SUBMISSION_ERROR %}
+        <h3 class="t5">{{ __('toplevel.ebulletin.responses.error.title') }}</h3>
+        <p>{{ __('toplevel.ebulletin.responses.error.body') }}</p>
+    {% else %}
+        <input type="hidden" name="_csrf" value="{{ csrfToken }}">
+        <input type="hidden" name="locale" value="{{ locale }}">
+
+        {{ formErrors(
+            errors = errors,
+            title = __('toplevel.ebulletin.responses.error.title')
+        ) }}
+
+        <fieldset class="form-fieldset">
+            <legend class="form-fieldset__legend u-visually-hidden">{{ __('toplevel.ebulletin.location')  }}</legend>
+            <div class="form-fieldset__fields">
+                {{ renderField({
+                    name: 'location',
+                    type: 'radio',
+                    options: [{
+                        label: __('toplevel.ebulletin.locations.england'),
+                        value: 'ENGLAND_MAIL'
+                    }, {
+                        label: __('toplevel.ebulletin.locations.wales'),
+                        value: 'WALES_MAIL'
+                    }, {
+                        label: __('toplevel.ebulletin.locations.scotland'),
+                        value: 'SCOTLAND_MAIL'
+                    }, {
+                        label: __('toplevel.ebulletin.locations.northernIreland'),
+                        value: 'NI_MAIL'
+                    }]
+                }) }}
+            </div>
+        </fieldset>
+
+        <fieldset class="form-fieldset">
+            <legend class="form-fieldset__legend u-visually-hidden">
+                {{ __('toplevel.ebulletin.yourDetails') }}
+            </legend>
+            <div class="form-fieldset__fields">
+                <div class="ff-field-pair">
+                    <div class="ff-field-pair__item">
+                        {{ renderField({
+                            type: 'text',
+                            name: 'firstName',
+                            label: __('toplevel.ebulletin.fields.firstName'),
+                            placeholder: __('toplevel.ebulletin.fields.firstName') + ' *',
```

Comment:
I wonder if this is better handled by checking `isRequired` in one of the field macros and appending the asterisk there?

Reply:
Good call